### PR TITLE
feat: extract credible sets and studies from all eQTL Catalogue finemapping results

### PR DESCRIPTION
**Pull request by ireneisdoomed**
This PR processes all eQTL Catalogue fine mapping results.
- We are now ingesting QTLs for all quantification methods: gene level, transcript level, transcript usage,... All of them are [described](https://www.ebi.ac.uk/eqtl/Methods/) here. 
-  Having more granular QTLs impact the `studyId` definition. We go from `publication_tissue_gene` to `publication_tissue_measuredtrait`. 
- The lead variant in the locus is now derived from the variant with the highest posterior probability. Finngen fine mapping results has also been changed to accommodate this.
- New `StudyLocus.calculate_credible_set_log10bf` method. This aggregates all single LBFs in a credible set.

The processing job takes 20 minutes. Full DAG will be ~50 minutes.

This is how the DAG looks like
image

### CODE REVIEW

File: src/gentropy/common/utils.py [hunk 1]
```diff
@@ -325,3 +327,24 @@ def parse_efos(efo_uri: Column) -> Column:
     """
     colname = efo_uri._jc.toString()
     return f.array_sort(f.expr(f"regexp_extract_all(`{colname}`, '([A-Z]+_[0-9]+)')"))
+
+
+def get_logsum(arr: NDArray[np.float64]) -> float:
```
[the remark this answers is not in the document]
Comment (d0choa) [reply src/gentropy/common/utils.py:332]: and probably make it private

File: src/gentropy/datasource/eqtl_catalogue/finemapping.py [hunk 2]
```diff
@@ -255,13 +253,59 @@ def from_susie_results(
...
+        """Load raw log Bayes Factors from eQTL Catalogue.
+
+        Args:
+            session (Session): Spark session.
+            lbf_path (str | list[str]): Path to raw table(s) containing Log Bayes Factors for each variant.
+
+        Returns:
+            DataFrame: Log Bayes Factors DataFrame.
+        """
+        return session.spark.read.csv(
```
Comment (d0choa) [inline src/gentropy/datasource/eqtl_catalogue/finemapping.py:306]: we clearly have different ways of understanding the world 🤣

File: src/gentropy/datasource/eqtl_catalogue/study_index.py [hunk 3]
```diff
@@ -40,24 +43,45 @@ class EqtlCatalogueStudyIndex:
             StructField("quant_method", StringType(), True),
         ]
     )
-    raw_studies_metadata_path = "https://raw.githubusercontent.com/eQTL-Catalogue/eQTL-Catalogue-resources/master/data_tables/dataset_metadata.tsv"
+    raw_studies_metadata_path = "https://raw.githubusercontent.com/eQTL-Catalogue/eQTL-Catalogue-resources/19929ff6a99bf402194292a14f96f9615b35f65f/data_tables/dataset_metadata.tsv"
```
Comment (d0choa) [inline src/gentropy/datasource/eqtl_catalogue/study_index.py:46]: nice

**Review by d0choa — APPROVED** [review]
Looks great. Think about the logsum and we can merge it.

File: src/gentropy/common/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ireneisdoomed) [reply src/gentropy/common/utils.py:332]: Thank you! As discussed, the StudyLocus incorporates a wrapper around this function that uses it in the specific context of calculating log10BFs for the credible set.
The same function is used differently for COLOC to calculate posterior probabilities for all H4 hypothesis, so not exactly StudyLocus specific.
We'll leave it like this

**Pull request merged**